src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java

```diff
@@ -260,8 +264,49 @@ private Model processModel(Model projectModel) throws IOException {
                 }
             }
 
-            // TODO
-            // update version within dependencies, dependency management, plugins, plugin management
+            for (Model previousModel : virtualProjectModelCache.values()) {
```

Comment:
there is no need to loop over the whole project cache for each project.
the processor is called for each project, so you need to do this only for the current project virtualProjectModel .

Reply:
And thank for your effort to update all locations of dependencies.

Reply:
For multimodules project me.qoomon.maven.gitversioning.GitVersioningModelProcessor#processModel(org.apache.maven.model.Model) is called per module. For each module, I am trying to update other poms which are referencing that one. If I do it once, it can cause persist stale data. For example X is parent module B,C,D are sub modules. If I update once per module, first module can not be updated. 
This can be simplified if I can I know which one is latest module.

Reply:
Ah I got it you are right. I will think about it at the weekend. I'll be back at monday.